### PR TITLE
Bugfix MTE-4495 Bitrise Xcode Check and Update workflow update

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-4495)

## :bulb: Description

The URL `https://app.bitrise.io/app/6c06d3a40422d10f/all_stack_info` is no longer available. (I'm not sure if it coincide with the new stack's release or not.) I found another way to query the current stack by heading over to the source of the Bitrise stack updates site (https://bitrise.io/stacks). The repo should reflect any updates to the stack accurately.

## :movie_camera: Demos

| Before | After |
| - | - |
|  |  |
|  |  |
|  |  |

<details>
<summary>Demo</summary>

</details>

## :pencil: Checklist
- [ ] I filled in the ticket numbers and a description of my work
- [ ] I updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] I ensured unit tests pass and wrote tests for new code
- [ ] If working on UI, I checked and implemented accessibility (Dynamic Text and VoiceOver)
- [ ] If adding telemetry, I read the [data stewardship requirements](https://github.com/mozilla-mobile/firefox-ios/wiki/Adding-Glean-Telemetry-Events) and will request a data review
- [ ] If needed, I updated documentation and added comments to complex code
- [ ] If needed, I added a backport comment (example `@Mergifyio backport release/v150.0`)

### CODE REVIEW

File: test-fixtures/update.py [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
 from ruamel.yaml import YAML
 
 
 import requests
-import semver
 from requests.exceptions import HTTPError
 
 
-BITRISE_STACK_INFO = 'https://app.bitrise.io/app/6c06d3a40422d10f/all_stack_info'
+BITRISE_STACK_INFO = 'https://api.github.com/repos/bitrise-io/stacks/contents/content/stack_reports'
```
Comment (clarmso) [inline test-fixtures/update.py:10]: I can fetch the info of this directory via Github's API.

https://github.com/bitrise-io/stacks/tree/main/content/stack_reports

File: test-fixtures/update.py [hunk 2]
```diff
@@ -29,22 +25,14 @@
 resp.raise_for_status()
 resp_json = resp.json()
 
-
-def parse_semver(raw_str):
```
Comment (clarmso) [inline test-fixtures/update.py:33]: This is not used.

File: test-fixtures/update.py [hunk 3]
```diff
@@ -69,16 +60,6 @@ def write_to_bitrise_yml(yaml, version):
         copyfile(tmp_file, BITRISE_YML)
         os.remove(tmp_file)
 
-def default_stack():
```
Comment (clarmso) [inline test-fixtures/update.py:72]: this is also not used.

**Comment by isabelrios** [comment]
@clarmso with this PR I saw by looking at bitrise [build](https://app.bitrise.io/build/fd140d7c-59cf-460e-bae0-94182b9182e8?tests_filter_status=failed) (it was taking more than 2hours) that focus build was waiting to create the shards for testing. Not sure about the error but in any case the value of the ENV variable should be false...
echo 'BUILD_FOCUS_IOS: true'
BUILD_FOCUS_IOS: true

there might be an edge case here where it did not update correctly (I think your PR to reduce the number of focus builds already landed right?)

**Comment by clarmso** [comment]
> @clarmso with this PR I saw by looking at bitrise [build](https://app.bitrise.io/build/fd140d7c-59cf-460e-bae0-94182b9182e8?tests_filter_status=failed) (it was taking more than 2hours) that focus build was waiting to create the shards for testing. Not sure about the error but in any case the value of the ENV variable should be false... echo 'BUILD_FOCUS_IOS: true' BUILD_FOCUS_IOS: true
> 
> there might be an edge case here where it did not update correctly (I think your PR to reduce the number of focus builds already landed right?)

The file `update.py` is under `test-fixtures/`. The step `determine_apps_affected` says we should compile focus as well if `test-fixtures/` is updated.
https://github.com/mozilla-mobile/firefox-ios/blob/main/bitrise.yml#L423

**Comment by clarmso** [comment]
> @clarmso with this PR I saw by looking at bitrise [build](https://app.bitrise.io/build/fd140d7c-59cf-460e-bae0-94182b9182e8?tests_filter_status=failed) (it was taking more than 2hours) that focus build was waiting to create the shards for testing. Not sure about the error but in any case the value of the ENV variable should be false... echo 'BUILD_FOCUS_IOS: true' BUILD_FOCUS_IOS: true
> 
> there might be an edge case here where it did not update correctly (I think your PR to reduce the number of focus builds already landed right?)

There is a separate `RUN_UI_TESTS` variable that examines if the source code has been updated. `run_firefox_ui_tests` uses `RUN_UI_TESTS` to determine if the smoke tests are run, but `focus_ui_test` uses `BUILD_FOCUS_IOS`.

Should we build firefox and focus for changes in `test-fixtures/` but not run the `ui_tests`?

File: bitrise.yml [hunk 4]
```diff
@@ -35,7 +35,7 @@ pipelines:
         - firefox_configure_build
       focus_ui_test:
         run_if:
-          expression: '{{ enveq "BUILD_FOCUS_IOS" "true" }}'
+          expression: '{{ enveq "RUN_UI_TESTS" "Run_UI_Tests" }}'
```
Comment (clarmso) [inline bitrise.yml:38]: Let's use `RUN_UI_TESTS` for all `ui_test` related workflows.
